### ISSUE TITLE
`automaticallyAdjustKeyboardInsets` Does Not adjust Insets When Keyboard Disappears again

### ISSUE THREAD

**Issue by MaxTheTurtle0** [facebook/react-native#47731]
### Description

When using `automaticallyAdjustKeyboardInsets={true}` in the new architecture, the insets are not adjusted properly. Initially, the content is pushed above the keyboard  if needed when it first appears, but the insets are not removed when the keyboard disappears. This leads to the `ScrollView` being permanently misaligned.

### Steps to reproduce

1. Clone the repository and cd into ReproducerApp
2. Install dependencies with `bun i`
3. cd into ios/ folder and install pods with fabric enabled `RCT_NEW_ARCH_ENABLED=1 pod install` (you can install with or without fabric enabled to compare behaviors)
4. `cd ..` and run `bun ios`

### React Native Version

0.76.2

### Affected Platforms

Runtime - iOS

### Areas

Fabric - The New Renderer

### Output of `npx react-native info`

```text
(node:54260) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
System:
  OS: macOS 15.1
  CPU: (8) arm64 Apple M1 Pro
  Memory: 751.11 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.11.0
    path: ~/.nvm/versions/node/v22.11.0/bin/node
  Yarn: Not Found
  npm:
    version: 10.9.0
    path: ~/.nvm/versions/node/v22.11.0/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.15.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2024.1 AI-241.15989.150.2411.11948838
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.11
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.2
    wanted: 0.76.2
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: Not found
  newArchEnabled: Not found
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

### Stacktrace or Logs

```text
.
```

### Reproducer

https://github.com/MaxTheTurtle0/adjustKeyboardInset-newArch

### New Arch (with Bug - keyboard insets don't get adjusted again when it disappears)
[video](https://github.com/user-attachments/assets/1489c04e-d045-4e5f-a046-34b8abc0ebb4)

### Without New Arch (everything works fine - the keyboard insets get adjusted again when it disappears)
[video](https://github.com/user-attachments/assets/81cdac5d-d959-45ec-9b7a-59ba4f7629d3)

**Comment by mbackonja** [comment]
Experiencing the same issue... I also have two recordings with possibly more details and I can see two different issues...

1. Create a form with 10 input fields inside a ScrollView.
2. Click on one of the first 4 input fields (those the keyboard wouldn't cover when it opens).
   - **Bug 1**: When the keyboard opens, it becomes impossible to scroll to the bottom of the form (the last 6 input fields and the button that are covered by the keyboard).
3. Dismiss the keyboard and then focus on one of the last 6 input fields (those that the keyboard would normally cover).
   - Scrolling works as expected while the keyboard is open (Bug 1 is not present here), but introduces a new issue when the keyboard is dismissed:
   - **Bug 2 (same as OP)**: An unnecessary bottom padding/margin remains, leaving extra space below the form and making the scrollbar look incorrect.

New architecture (broken):

https://github.com/user-attachments/assets/14c189db-c9f3-4c02-b03d-6e48be01cb98

Old architecture (working):

https://github.com/user-attachments/assets/10965427-2464-496e-b99d-e941118bea34

**Comment by zhongwuzw** [comment]
May fixed in #47924 .
